File: backend-java/src/main/java/app/healthfamily/domain/hospital/Hospital.java
package app.healthfamily.domain.hospital;

import app.healthfamily.domain.shared.DomainException;
import app.healthfamily.domain.shared.OwnedResource;
import java.util.Optional;

/**
 * かかりつけ病院。
 *
 * <p>ドメイン規則は所有権と名前の必須のみ。判断が増えたらここに足す。
 */
public record Hospital(
        String id,
        String userId,
        String name,
        String hospitalType,
        String address,
        String phoneNumber,
        String department,
        String doctorName,
        String notes)
        implements OwnedResource {

    public Hospital {
        if (id == null || id.isBlank()) {
            throw DomainException.validation("病院のIDは必須です");
        }
        if (name == null || name.isBlank()) {
            throw DomainException.validation("病院の名前は必須です");
        }
        name = name.trim();
    }

    @Override
    public String ownerId() {
        return userId;
    }

    public Optional<String> departmentName() {
        return Optional.ofNullable(department);
    }
}

File: frontend/src/entities/hospital/model/hospital.ts
// 病院の部分更新に使う入力。カード内の編集 UI (entities) と更新 feature の双方が
// 参照するため、依存が上向きにならないよう entities 側に置く。
export interface UpdateHospitalInput {
  name?: string;
  address?: string;
  phone?: string;
  department?: string;
  doctorName?: string;
  notes?: string;
}
